data/refprog/reference-program/src/main/java/com/farao_community/farao/data/refprog/reference_program/ReferenceProgram.java

```diff
@@ -19,56 +16,60 @@
  */
 public class ReferenceProgram {
     private final List<ReferenceExchangeData> referenceExchangeDataList;
-    private final Set<Country> countries;
-    private final Map<Country, Double> netPositions;
+    private final Set<ReferenceProgramArea> referenceProgramAreas;
+    private final Map<ReferenceProgramArea, Double> netPositions;
 
     public ReferenceProgram(List<ReferenceExchangeData> referenceExchangeDataList) {
         this.referenceExchangeDataList = referenceExchangeDataList;
-        this.countries = new HashSet<>();
+        this.referenceProgramAreas = new HashSet<>();
         this.referenceExchangeDataList.stream().forEach(referenceExchangeData -> {
-            if (referenceExchangeData.getAreaOut() != null) {
-                countries.add(referenceExchangeData.getAreaOut());
+            referenceProgramAreas.add(referenceExchangeData.getAreaOut());
+            referenceProgramAreas.add(referenceExchangeData.getAreaIn());
+            if (referenceExchangeData.getAreaOut() != null && !referenceExchangeData.getAreaOut().isVirtualHub()) {
+                referenceProgramAreas.add(referenceExchangeData.getAreaOut());
             }
-            if (referenceExchangeData.getAreaIn() != null) {
-                countries.add(referenceExchangeData.getAreaIn());
+            if (referenceExchangeData.getAreaIn() != null && !referenceExchangeData.getAreaIn().isVirtualHub()) {
+                referenceProgramAreas.add(referenceExchangeData.getAreaIn());
             }
         });
-        netPositions = new EnumMap<>(Country.class);
-        countries.forEach(country -> netPositions.put(country, computeGlobalNetPosition(country)));
+        netPositions = new HashMap<>();
+        referenceProgramAreas.forEach(country -> netPositions.put(country, computeGlobalNetPosition(country)));
     }
 
     public List<ReferenceExchangeData> getReferenceExchangeDataList() {
         return referenceExchangeDataList;
     }
 
-    public Set<Country> getListOfCountries() {
-        return countries;
+    public Set<ReferenceProgramArea> getListOfCountries() {
```

Comment:
Could you rename this method : getListOfAreas() ?

Reply:
sure!